app/controllers/runs_controller.rb

```diff
@@ -128,4 +130,8 @@ def handle_first_visit
       redirect_to cant_parse_path
     end
   end
+
+  def set_filetype
+    @filetype = {'livesplit' => 'lss', 'urn' => 'json'}.fetch(params[:program], params[:program])
+  end
```

Comment:
Instead of putting this as a before_action in the controller, can we put it in a method on the `Run` model? In fact, can we just make it a `filename` method that returns the whole string `"#{to_param}.#{extension}"`? I think this would be cleaner than adding another method down here, and would support future cases where we need to give the file a name in another controller (e.g. #57).

This was a good call refactoring this into its own method, it was getting messy even without adding Urn.